diff -up boost-1.41.0.cmake0/boost/interprocess/detail/move.hpp\~ boost-1.41.0.cmake0/boost/interprocess/detail/move.hpp
--- boost-1.41.0.cmake0/boost/interprocess/detail/move.hpp~	2009-11-18 17:44:37.000000000 +0100
+++ boost-1.41.0.cmake0/boost/interprocess/detail/move.hpp	2015-01-14 00:17:47.065934278 +0100
@@ -339,7 +339,7 @@ rvalue_reference move (input_reference);
 #else
 template <class T> inline 
 typename remove_reference<T>::type&& move(T&& t)
-{  return t;   }
+{ return static_cast<typename remove_reference<T>::type &&>(t); }
 #endif
 
 //////////////////////////////////////////////////////////////////////////////

Diff finished.  Wed Jan 14 00:17:50 2015
