--- boost/thread/detail/thread.hpp
+++ boost/thread/detail/thread.hpp
@@ -40,6 +40,9 @@ namespace boost
         {
         public:
 #ifdef BOOST_HAS_RVALUE_REFS
+            thread_data(const F& f_):
+                f(f_)
+            {}
             thread_data(F&& f_):
                 f(static_cast<F&&>(f_))
             {}
@@ -346,7 +349,7 @@ namespace boost
 #ifdef BOOST_HAS_RVALUE_REFS
     inline thread&& move(thread&& t)
     {
-        return t;
+        return static_cast<thread&&>(t);
     }
 #else
     inline detail::thread_move_t<thread> move(detail::thread_move_t<thread> t)
