diff -up boost-1.41.0.cmake0/boost/config/stdlib/libstdcpp3.hpp\~ boost-1.41.0.cmake0/boost/config/stdlib/libstdcpp3.hpp
--- boost-1.41.0.cmake0/boost/config/stdlib/libstdcpp3.hpp~	2009-11-18 17:44:37.000000000 +0100
+++ boost-1.41.0.cmake0/boost/config/stdlib/libstdcpp3.hpp	2014-04-29 00:53:57.975139925 +0200
@@ -31,7 +31,8 @@
 
 #ifdef __GLIBCXX__ // gcc 3.4 and greater:
 #  if defined(_GLIBCXX_HAVE_GTHR_DEFAULT) \
-        || defined(_GLIBCXX__PTHREADS)
+        || defined(_GLIBCXX__PTHREADS) \
+        || defined(_GLIBCXX_HAS_GTHREADS)
       //
       // If the std lib has thread support turned on, then turn it on in Boost
       // as well.  We do this because some gcc-3.4 std lib headers define _REENTANT

Diff finished.  Tue Apr 29 00:54:01 2014
