--- ./libs/mpi/src/CMakeLists.txt	2009-11-18 17:44:37.000000000 +0100
+++ ./libs/mpi/src/CMakeLists.txt	2012-03-09 19:15:05.365549067 +0100
@@ -68,7 +71,7 @@ if (NOT APPLE)
       python/status.cpp
       python/py_timer.cpp
 
-      DEPENDS boost_python boost_mpi
+      DEPENDS boost_python boost_mpi boost_mpi_python boost_serialization
       SHARED_COMPILE_FLAGS "-DBOOST_MPI_DYN_LINK=1 -DBOOST_MPI_PYTHON_DYN_LINK=1 -DBOOST_PYTHON_DYN_LINK=1"
       LINK_FLAGS ${MPI_LINK_FLAGS}
       LINK_LIBS ${MPI_LIBRARIES} ${PYTHON_LIBRARIES}
--- ./tools/build/CMake/BoostCore.cmake	2012-03-08 22:18:12.463206955 +0100
+++ ./tools/build/CMake/BoostCore.cmake	2012-03-09 19:16:40.238345008 +0100
@@ -1491,19 +1491,19 @@ endmacro(boost_add_executable)
 #  Macro for building boost.python extensions
 #
 macro(boost_python_extension MODULE_NAME)
-  parse_arguments(BPL_EXT 
-    "" 
-    "" 
-    ${ARGN})
-  
+  parse_arguments(BPL_EXT
+    "DEPENDS;COMPILE_FLAGS;LINK_FLAGS;LINK_LIBS;${BOOST_ADD_ARG_NAMES}"
+    "${BOOST_ADDLIB_OPTION_NAMES}"
+    ${ARGN}
+    )
+
   boost_add_single_library(
     ${MODULE_NAME}
     ${BPL_EXT_DEFAULT_ARGS}
     MODULE
-    LINK_LIBS ${PYTHON_LIBRARIES}
-    DEPENDS boost_python
+    LINK_LIBS ${PYTHON_LIBRARIES} ${BPL_EXT_LINK_LIBS}
+    DEPENDS boost_python ${BPL_EXT_DEPENDS}
     SHARED
     MULTI_THREADED
-    DEBUG
     )
 endmacro()
