diff -up boost-1.41.0.cmake0/boost/smart_ptr/weak_ptr.hpp\~ boost-1.41.0.cmake0/boost/smart_ptr/weak_ptr.hpp
--- boost-1.41.0.cmake0/boost/smart_ptr/weak_ptr.hpp~	2009-11-18 17:44:37.000000000 +0100
+++ boost-1.41.0.cmake0/boost/smart_ptr/weak_ptr.hpp	2015-01-14 21:38:05.940249719 +0100
@@ -76,7 +76,18 @@ public:
 
 #if defined( BOOST_HAS_RVALUE_REFS )
 
-    template<class Y>
+    weak_ptr( weak_ptr const & r ) : px( r.px ), pn( r.pn )
+    {
+    }
+
+    weak_ptr & operator=( weak_ptr const & r )
+    {
+        px = r.px;
+        pn = r.pn;
+        return *this;
+    }
+
+  template<class Y>
 #if !defined( BOOST_SP_NO_SP_CONVERTIBLE )
 
     weak_ptr( weak_ptr<Y> && r, typename detail::sp_enable_if_convertible<Y,T>::type = detail::sp_empty() )

Diff finished.  Wed Jan 14 21:44:59 2015
